api/desecapi/pdns.py

```diff
@@ -108,3 +133,18 @@ def set_dyn_records(name, a, aaaa):
             _delete_or_replace_rrset(name, 'aaaa', aaaa),
         ]
     })
+
+
+def set_rrset(zone, name, type, value):
```

Comment:
The `name` and `zone` argument seem redundant, or I could not figure out why they ever should be set to different values. Cf. code comment below: `... for the zone with the given name`.

The `type` argument shadows the python3 built-in function `type()`, which should be avoided. Suggestion: `rr_type`. (This is also true for already existing `type` arguments in the code.)